recipes-extended/xen/xen-ocaml-libs.bb

```diff
@@ -0,0 +1,110 @@
+require recipes-extended/xen/xen.inc
+require xen-common.inc
+
+inherit findlib
+
+DESCRIPTION = "Xen hypervisor ocaml libs and xenstore components"
+
+# OpenXT packages both the C and OCaml versions of XenStored.
+# This recipe packages the OCaml daemon; xen.bb packages the C one.
+FILES_xen-xenstored-ocaml = " \
+    ${sbindir}/xenstored.xen-xenstored-ocaml \
+    ${localstatedir}/lib/xenstored \
+    ${sysconfdir}/init.d/xenstored.xen-xenstored-ocaml \
+    ${sysconfdir}/xen/oxenstored.conf \
+    "
+
+PROVIDES =+ "xen-xenstored xen-xenstored-ocaml"
+RPROVIDES_xen-xenstored-ocaml = "xen-xenstored xen-xenstored-ocaml"
+
+DEPENDS += " \
+    util-linux \
+    xen \
+    xen-blktap \
+    libnl \
+    ocaml-cross \
+    "
+EXTRA_OECONF_remove = "--disable-ocamltools"
+
+SRC_URI_append = " \
+    file://xenstored.initscript \
+    file://oxenstored.conf \
+    "
+
+PACKAGES = " \
+    xen-xenstored-ocaml \
+    ${PN}-dev \
+    ${PN}-dbg \
+    ${PN}-staticdev \
+    ${PN} \
+    "
+
+FILES_${PN}-dev = "${ocamllibdir}/*/*.so"
+FILES_${PN}-dbg += "${ocamllibdir}/*/.debug/*"
+FILES_${PN}-staticdev = "${ocamllibdir}/*/*.a"
+FILES_${PN} = "${ocamllibdir}/*"
+
+EXTRA_OEMAKE += "CROSS_SYS_ROOT=${STAGING_DIR_HOST} CROSS_COMPILE=${HOST_PREFIX}"
+EXTRA_OEMAKE += "CONFIG_IOEMU=n"
+EXTRA_OEMAKE += "DESTDIR=${D}"
+
+#Make sure we disable all compiler optimizations to avoid a nasty segfault in the 
+#reboot case.
+BUILD_LDFLAGS += " -Wl,-O0 -O0"
+BUILDSDK_LDFLAGS += " -Wl,-O0 -O0"
+TARGET_LDFLAGS += " -Wl,-O0 -O0"
+BUILD_OPTIMIZATION = "-pipe"
+FULL_OPTIMIZATION = "-pipe ${DEBUG_FLAGS}"
+
+TARGET_CC_ARCH += "${LDFLAGS}"
+CC_FOR_OCAML="i686-oe-linux-gcc"
+
+INITSCRIPT_PACKAGES = "xen-xl xen-xenstored-ocaml"
+INITSCRIPT_NAME_xen-xenstored-ocaml = "xenstored"
+INITSCRIPT_PARAMS_xen-xenstored-ocaml = "defaults 05"
+
+pkg_postinst_xen-xenstored-ocaml () {
+    update-alternatives --install ${sbindir}/xenstored xenstored xenstored.xen-xenstored-ocaml 100
+    update-alternatives --install ${sysconfdir}/init.d/xenstored xenstored-initscript xenstored.xen-xenstored-ocaml 100
+}
+
+pkg_prerm_xen-xenstored-ocaml () {
+    update-alternatives --remove xenstored xenstored.xen-xenstored-ocaml
+    update-alternatives --remove xenstored-initscript xenstored.xen-xenstored-ocaml
+}
+
+do_configure_prepend() {
+	#remove optimizations in the config files
+	sed -i 's/-O2//g' ${WORKDIR}/xen-${XEN_VERSION}/Config.mk
+	sed -i 's/-O2//g' ${WORKDIR}/xen-${XEN_VERSION}/config/StdGNU.mk
+}
```

Comment:
The `do_configure_prepend` function again originated in the libxl recipe and should be removable from this recipe.